src/tts_text/phoneme_covering.py

```diff
@@ -0,0 +1,194 @@
+    words = document_text.split(" ")
+
+    counter = Counter(words)
+    for language, phoneme_list in phonemes.items():
+        phoneme_count = 0
+        found_phonemes = []
+        found_example_words = []
+        for phoneme in phoneme_list:
+            for word in phoneme["examples"]:
+                occurences = counter[word]
+                phoneme_count += occurences
+                if occurences > 0:
+                    found_phonemes.append(phoneme["name"])
+                    found_example_words.append(word)
+        document[f"{language}_phoneme_count"] = phoneme_count
+        document[f"{language}_phonemes"] = found_phonemes
+        document[f"{language}_found_example_words"] = found_example_words
+        document[f"{language}_unique_phonemes"] = set(found_phonemes)
+        document[f"{language}_unique_phonemes_count"] = len(set(found_phonemes))
+
+    document["all_phoneme_count"] = (
+        document["da_phoneme_count"] + document["en_phoneme_count"]
+    )
+    document["all_phonemes"] = document["da_phonemes"] + document["en_phonemes"]
+    document["all_found_example_words"] = (
+        document["da_found_example_words"] + document["en_found_example_words"]
+    )
+    document["all_unique_phonemes"] = set(
+        document["da_unique_phonemes"] | document["en_unique_phonemes"]
+    )
+    document["all_unique_phonemes_count"] = len(document["all_unique_phonemes"])
+    return document
+
+
+def build_phoneme_covering_dataset(
+    cfg: DictConfig, output_dir: Path | str
+) -> list[str]:
+    """Create a phoneme covering set from the sorted wiki dataset.
+
+    A phoneme covering set is a set of strings that contains at least one example
+    word for each phoneme.
+
+    Args:
+        cfg: The Hydra configuration object.
+        output_dir: The directory to save the dataset to.
+
+    Returns:
+        The phoneme covering set.
+    """
+    sorted_dataset = sort_wiki_by_phoneme_occurence(cfg=cfg)
+
+    phoneme_path = Path(cfg.dirs.phoneme_dir)
+    with phoneme_path.open() as f:
+        phonemes = json.load(f)
+    all_phone_names = {entry["name"] for entry in phonemes["da"]}.union(
+        {entry["name"] for entry in phonemes["en"]}
+    )
+    covering_set = []
+    for document in tqdm(sorted_dataset):
+        if not all_phone_names:
+            break
+        document_phonemes = document["all_phonemes"]
+        new_phonemes = set(all_phone_names).intersection(document_phonemes)
+        for new_phoneme in new_phonemes:
+            all_phone_names.remove(new_phoneme)
+        if new_phonemes:
+            covering_set.append(document)
+
+    covering_set_dataset = Dataset.from_list(covering_set)
+    get_example_words(phonemes=phonemes)
+
+    def extract_paragraph_with_example_word(document: dict) -> dict:
+        found_paragraphs = []
+        for paragraph in re.split(
+            "\t|\n|_NEWLINE_|_START_ARTICLE_|_START_SECTION_|_START_PARAGRAPH_",
```

Comment:
You're using this twice in here. Maybe make it a global constant, or part of the config as a list of strings, that you join with pipes here?